Google.PowerShell.IntegrationTests/Compute/Compute.GceHealthCheck.Tests.ps1

```diff
@@ -0,0 +1,238 @@
+            { Get-GceHealthCheck $healthCheckName -Http } | Should Throw 404
+        }
+
+        It "should use object pipeline" {
+            Get-GceHealthCheck $healthCheckName -Http |
+                Remove-GceHealthCheck
+            { Get-GceHealthCheck $healthCheckName -Http } | Should Throw 404
+        }
+
+        It "should fail removing HTTPS health check" {
+            { Remove-GceHealthCheck $healthCheckName -Https } | Should Throw 404
+            Remove-GceHealthCheck $healthCheckName -Http
+        }
+    }
+
+    Context "Remove Https" {
+        BeforeEach {
+            Add-GceHealthCheck $healthCheckName -Https
+        }
+
+        It "should work" {
+            Remove-GceHealthCheck $healthCheckName -Https
+            { Get-GceHealthCheck $healthCheckName -Https } | Should Throw 404
+        }
+
+        It "should use object pipeline" {
+            Get-GceHealthCheck $healthCheckName -Https |
+                Remove-GceHealthCheck
+            { Get-GceHealthCheck $healthCheckName -Https } | Should Throw 404
+        }
+
+        It "should fail removing HTTP health check" {
+            { Remove-GceHealthCheck $healthCheckName -Http } | Should Throw 404
+            Remove-GceHealthCheck $healthCheckName -Https
+        }
+    }
+}
+
+Describe "Set-GceHealthCheck" {
+    $healthCheckName = "test-set-gcehealthcheck-$r"
+
+    Add-GceHealthCheck $healthCheckName
+    Add-GceHealthCheck $healthCheckName -Https
+
+    AfterAll {
+        Get-GceHealthCheck |
+            Remove-GceHealthCheck
+    }
+
+    It "should work" {
+        $healthChecks = Get-GceHealthCheck $healthCheckName | %{
+            $_.Port = 50
+            $_.Host = "google.com"
+            $_.RequestPath = "/some/path"
+            $_.CheckIntervalSec = 2
+            $_.TimeoutSec = 2
+            $_.HealthyThreshold = 3
+            $_.UnhealthyThreshold = 3
+            $_
+        } | Set-GceHealthCheck
+
+        $healthChecks.Count | Should Be 2
+        $healthChecks.Name | Should Be $healthCheckName
+        $healthChecks.Port | Should Be 50
+        $healthChecks.Host | Should Be "google.com"
+        $healthChecks.RequestPath | Should Be "/some/path"
+        $healthChecks.CheckIntervalSec | Should Be 2
+        $healthChecks.TimeoutSec | Should Be 2
+        $healthChecks.HealthyThreshold | Should Be 3
+        $healthChecks.UnhealthyThreshold | Should Be 3
+
+        $healthChecks = Get-GceHealthCheck $healthCheckName
+
+        $healthChecks.Count | Should Be 2
+        $healthChecks.Name | Should Be $healthCheckName
+        $healthChecks.Port | Should Be 50
+        $healthChecks.Host | Should Be "google.com"
+        $healthChecks.RequestPath | Should Be "/some/path"
+        $healthChecks.CheckIntervalSec | Should Be 2
+        $healthChecks.TimeoutSec | Should Be 2
+        $healthChecks.HealthyThreshold | Should Be 3
+        $healthChecks.UnhealthyThreshold | Should Be 3
+    }
+}
```

Comment:
This test calls `Set-GCloudConfig` but does not restore the Cloud configuration at the end.